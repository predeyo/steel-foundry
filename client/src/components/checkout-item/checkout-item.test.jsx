import React from "react";
import { shallow } from "enzyme";
import { CheckoutItem } from "./checkout-item.component.jsx";

describe("CheckoutItem component", () => {
  let wrapper;
  let mockCartItem = {
    name: "Red hat",
    price: "99",
    imageUrl: "www.testimage.com"
  };
  let mockAddItem;
  let mockRemoveItem;
  let mockClearItem;
  beforeEach(() => {
    mockAddItem = jest.fn();
    mockRemoveItem = jest.fn();
    mockClearItem = jest.fn();
    let mockProps = {
      cartItem: mockCartItem,
      addItem: mockAddItem,
      removeItem: mockRemoveItem,
      clearItem: mockClearItem
    };
    wrapper = shallow(<CheckoutItem {...mockProps} />);
  });

  it("should render CheckoutItem component", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("shold call addItem when increase item count is clicked", () => {
    wrapper.find(".add-item").simulate("click");

    expect(mockAddItem).toHaveBeenCalled();
  });

  it("shold call removeItem when decrese item count is clicked", () => {
    wrapper.find(".remove-item").simulate("click");

    expect(mockRemoveItem).toHaveBeenCalled();
  });

  it("shold call clearItem when clear-button is clicked", () => {
    wrapper.find(".clear-button").simulate("click");

    expect(mockClearItem).toHaveBeenCalled();
  });
});
