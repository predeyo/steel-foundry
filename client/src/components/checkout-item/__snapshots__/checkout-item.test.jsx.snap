// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem component should render CheckoutItem component 1`] = `
"<div className=\\"checkout-item\\">
  <div className=\\"image-container\\">
    <img src=\\"www.testimage.com\\" alt=\\"item\\" />
  </div>
  <span className=\\"name\\">
    Red hat
  </span>
  <span className=\\"quantity\\">
    <span className=\\"arrow remove-item\\" onClick={[Function: onClick]}>
      ❮
    </span>
    <span className=\\"value\\" />
    <span className=\\"arrow add-item\\" onClick={[Function: onClick]}>
      ❯
    </span>
  </span>
  <span className=\\"price\\">
    99
  </span>
  <div className=\\"clear-button\\" onClick={[Function: onClick]}>
    ✕
  </div>
</div>"
`;
